src/test/ui/impl-trait/impl-generic-mismatch.stderr

```diff
@@ -0,0 +1,44 @@
+error[E0643]: method `foo` has incompatible signature for trait
+  --> $DIR/impl-generic-mismatch.rs:18:12
+   |
+LL |     fn foo(&self, _: &impl Debug);
+   |                       ---------- declaration in trait here
+...
+LL |     fn foo<U: Debug>(&self, _: &U) { }
+   |            ^ expected `impl Trait`, found generic parameter
+help: try removing the generic parameter and using `impl Trait` instead
+   |
+LL |     fn foo(&self, _: &impl Debug) { }
+   |
+
+error[E0643]: method `bar` has incompatible signature for trait
+  --> $DIR/impl-generic-mismatch.rs:27:23
+   |
+LL |     fn bar<U: Debug>(&self, _: &U);
+   |            - declaration in trait here
+...
+LL |     fn bar(&self, _: &impl Debug) { }
+   |                       ^^^^^^^^^^ expected generic parameter, found `impl Trait`
+help: try changing the `impl Trait` argument to a generic parameter
+   |
+LL |     fn bar<U: Debug><U: Debug>(&self, _: &U);
```

Comment:
Incorrect span for the suggestion. This should be in the `bar` for the `impl Bar for ()`